api/utils/sshutils/ssh.go

```diff
@@ -250,3 +260,263 @@ func KeysEqual(ak, bk ssh.PublicKey) bool {
+	// for some reason what PuTTY names 'P' is Primes[1] to Go, and what PuTTY names 'Q' is Primes[0] to Go
+	P := privateKey.Primes[1]
+	mpintP, err := getRFC4251Mpint(P)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	err = binary.Write(ppkPrivateKey, binary.BigEndian, mpintP)
+	if err != nil {
+		return nil, trace.Wrap(fmt.Errorf("failed to write P to buffer: %T", err))
+	}
+
+	// mpint: the other prime factor q of the key. (RSA keys stored in this format are expected to have exactly two prime factors.)
+	// this is known as 'Q'
+	Q := privateKey.Primes[0]
+	mpintQ, err := getRFC4251Mpint(Q)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	err = binary.Write(ppkPrivateKey, binary.BigEndian, mpintQ)
+	if err != nil {
+		return nil, trace.Wrap(fmt.Errorf("failed to write Q to buffer: %T", err))
+	}
+
+	// mpint: the multiplicative inverse of q modulo p.
+	// this is known as 'iqmp'
+	iqmp := new(big.Int).ModInverse(Q, P)
+	mpintIQMP, err := getRFC4251Mpint(iqmp)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	err = binary.Write(ppkPrivateKey, binary.BigEndian, mpintIQMP)
+	if err != nil {
+		return nil, trace.Wrap(fmt.Errorf("failed to write iqmp to buffer: %T", err))
+	}
+
+	// now we need to base64-encode the PPK-formatted private key which is made up of the above values
+	ppkPrivateKeyBase64 := make([]byte, base64.StdEncoding.EncodedLen(ppkPrivateKey.Len()))
+	base64.StdEncoding.Encode(ppkPrivateKeyBase64, ppkPrivateKey.Bytes())
+
+	// build Teleport public key path
+	// fortunately, this is the one thing that's in exactly the same format that the PPK file uses, so we just copy it verbatim
+	teleportPublicKey := string(pub)
+	// remove ssh-rsa from beginning of string if present
+	if !strings.HasPrefix(teleportPublicKey, "ssh-rsa") {
+		return nil, trace.Wrap(fmt.Errorf("pub does not appear to be an ssh-rsa public key: %T", err))
+	}
+	teleportPublicKey = strings.TrimSuffix(strings.TrimPrefix(teleportPublicKey, "ssh-rsa "), "\n")
+	// we use the trimmed public key from here on out
+	teleportPublicKeyBytes := []byte(teleportPublicKey)
+
+	// the PPK file contains an anti-tampering MAC which is made up of various values which appear in the file.
+	// copied from Section C.3 of https://the.earth.li/~sgtatham/putty/0.76/htmldoc/AppendixC.html#ppk:
+	// hex-mac-data is a hexadecimal-encoded value, 64 digits long (i.e. 32 bytes), generated using the HMAC-SHA-256 algorithm with the following binary data as input:
+	// string: the algorithm-name header field.
+	// string: the encryption-type header field.
+	// string: the key-comment-string header field.
+	// string: the binary public key data, as decoded from the base64 lines after the 'Public-Lines' header.
+	// string: the plaintext of the binary private key data, as decoded from the base64 lines after the 'Private-Lines' header.
+
+	// these values are also used in the MAC generation, so we declare them as variables
+	keyType := "ssh-rsa"
```

Comment:
Would this make sense as a constant? Looks like we use the `"ssh-rsa"` string a number of times.